crates/core/tedge_api/src/entity_store.rs

```diff
@@ -161,6 +163,7 @@ impl EntityStore {
         external_id_validator_fn: SF,
         telemetry_cache_size: usize,
         log_dir: P,
+        clean_start: bool,
```

Comment:
Passing a `log_dir` and `clean_start` flag telling to truncate or not the log.

I would consider to pass only a `MessageLogWriter`, built from the log using  `MessageLogWriter::new(log_dir)` or  `MessageLogWriter::new_truncated(log_dir)`:

```suggestion
        message_log: MessageLogWriter,
```

For that to work the `MessageLogWriter` has to be improved to build a `MessageLogReader` on the same log dir. So the entity store can use its  `self.message_log` for reads and writes.

It might be better to address this only a follow-up PR after the 1.0 release.

Reply:
One suggestion is to have a `MessageLog` abstraction, which is passed to the entity store, out of which the `MessageLogWriter` and `MessageLogReader` instances can be retrieved.